### PR TITLE
Creating clusters cannot be deleted

### PR DESCRIPTION
**Pull request by akarukappadath**
\<your comments for this PR go here\>

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by akarukappadath** [comment]
jenkins retest

**Comment by akarukappadath** [comment]
jenkins retest

File: src/test/scala/org/broadinstitute/dsde/workbench/leonardo/service/AuthProviderSpec.scala [hunk 1]
```diff
@@ -89,6 +89,9 @@ class AuthProviderSpec extends FreeSpec with ScalatestRouteTest with Matchers wi
       val syncRequest = HttpRequest(POST, Uri(s"/notebooks/$googleProject/$clusterName/api/localize"))
       proxy.proxyLocalize(userInfo, GoogleProject(googleProject), ClusterName(clusterName), syncRequest).futureValue
 
+      // change cluster status to Running so that it can be deleted
+      dbFutureValue { _ => DbSingleton.ref.dataAccess.clusterQuery.setToRunning(cluster1.id, IP("numbers.and.dots"))}
```
Comment (rtitle) [inline src/test/scala/org/broadinstitute/dsde/workbench/leonardo/service/AuthProviderSpec.scala]: Minor, can do:
```
dbFutureValue { _.clusterQuery.setToRunning(cluster1.id, IP("numbers.and.dots")) }
```

File: automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala [hunk 2]
```diff
@@ -422,6 +422,17 @@ trait LeonardoTestUtils extends WebBrowserSpec with Matchers with Eventually wit
     val testResult: Try[T] = Try {
       testCode(cluster)
     }
+
```
Comment (rtitle) [inline automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala]: Just thinking out loud here. If `monitorCreate` is true, then the sequence will be:
1. Create
2. Wait till Running
3. Run test code
4. Delete

When `monitorCreate` is false, the sequence will be:
1. Create
2. Run test code
3. Wait till Running or Error
4. Delete

Yeah, I think this makes sense -- I like how we tolerate Error status in the second scenario. 👍

**Comment by akarukappadath** [comment]
jenkins retest

**Pull request merged**